src/Transmogrification.cpp

```diff
@@ -470,13 +470,62 @@ bool Transmogrification::CanTransmogrifyItemWithItem(Player* player, ItemTemplat
         {
             if (source->Class == ITEM_CLASS_ARMOR && !AllowMixedArmorTypes)
                 return false;
-            if (source->Class == ITEM_CLASS_WEAPON && !AllowMixedWeaponTypes)
-                return false;
+            if (source->Class == ITEM_CLASS_WEAPON)
+            {
+                if (AllowMixedWeaponTypes == MIXED_WEAPONS_STRICT)
+                {
+                    return false;
+                }
+                else if (AllowMixedWeaponTypes == MIXED_WEAPONS_MODERN)
```

Comment:
if you have an if condition with return, it is not necessary the `else`
```suggestion
                if (AllowMixedWeaponTypes == MIXED_WEAPONS_MODERN)
```